event-store-plans.md

```diff
@@ -0,0 +1,174 @@
+
+```csharp
+public interface IEventSession : IEventStore
+{
+    Task SaveChangesAsync(CancellationToken token = default);
+}
+```
+
+It'd be the `DocumentSession`, but with a more event store friendly interface so you don't have to do `session.Events.******` on everything. 
+
+
+## Document by document type identity map behavior
+
+Today, document sessions either have the identity map behavior for all document types or no identity map for all types. I think especially around the 
+projections, that it would be helpful to selectively turn on the identity map within the session for certain document types. I think this is actually pretty
+low hanging fruit because all you'd do is switch up the cached `IDocumentStorage` objects for a certain document type.
+
+
+## Ensure that aggregates updated "Inline" are coming from identity map
+
+Imagine this very common scenario:
+
+A CQRS command handler needs to:
+
+1. Fetch an aggregate of an event stream as a "write model" to process the command
+2. Emit some events
+3. Maybe update the projected aggregation if it's running `Inline`
+4. For whatever reason, access the current state of the aggregate again in the handler. Maybe because it's a response type in an HTTP handler, maybe to transmit
+   messages, who knows
+
+Maybe you even want the updated aggregate to make some kind of sense of the next part of the work -- which I'd at least push Wolverine folks to do with cascading messages, but still
+
+Regardless of the Marten `IDocumentSession` type, I think it would be valuable to **prevent** unnecessary extra reads of the aggregate, especially if there's some
+re-application of the events. 
+
+At a minimum, I think that the aggregation fetching should use the identity map *even if the session itself isn't using the identity map*. See the next item.
+This probably impacts both the `FetchForWriting()` behavior and the inline projection behavior
+
+This absolutely impacts the Wolverine "aggregate handler workflow". There's an opportunity to eliminate some serious inefficiency today
+
+
+## Versioned Projections for Blue/Green deployment
+
+This is really just sticking a "projection version number" on the Projection subclasses that would potentially be used as a suffix on the document
+tables for the projected documents (the document alias). This would enable blue/green deployment of the same projection so that nodes on the previous version can be 
+using the previous revision/definition of the projection while other nodes are running the newer model. 
+
+Obviously there's some serious concern about just how the new projection gets rebuilt in time. I think the "catch up" mode that this document
+introduces as an "Async strategy for FetchForWriting()" is necessary to make this viable
+
+
+## "Poor Man's Load Balancing" of the Async Daemon
+
+See Oskar's post [How to scale out Marten](https://event-driven.io/en/scaling_out_marten/). I'm voting to just bake this work directly into Marten as a 
```

Comment:
Do you think about keeping it as sharding, as I described or adding also something more to that? The presented approach assumed that each async daemon instance has a different set of projections. It could be useful to think if we could give option to specify the projection set of active projections through the environment variables or config, then one can use the same artefact (e.g. docker image) but provide different settings to decide on what to handle.